src/tests/class_tests/openms/source/Ms2IdentificationRate_test.cpp

```diff
@@ -0,0 +1,223 @@
+
+FeatureMap fmap_empty;
+
+
+fmap.setUnassignedPeptideIdentifications(pep_ids);
+
+//construct MSSpectrum
+MSSpectrum ms2_1;
+ms2_1.setMSLevel(2);
+MSSpectrum ms2_2;
+ms2_2.setMSLevel(2);
+MSSpectrum ms2_3;
+ms2_3.setMSLevel(2);
+MSSpectrum ms2_4;
+ms2_4.setMSLevel(2);
+MSSpectrum ms2_5;
+ms2_5.setMSLevel(2);
+MSSpectrum ms2_6;
+ms2_6.setMSLevel(2);
+MSSpectrum ms1;
+ms1.setMSLevel(1);
+
+std::vector<MSSpectrum> ms_spectra = {ms2_1, ms2_2, ms2_3, ms2_4, ms2_5, ms2_6, ms1};
+std::vector<MSSpectrum> ms1_spectra = {ms1};
+std::vector<MSSpectrum> ms2_2_spectra = {ms2_1};
+
+//construct MSExperiment
+MSExperiment ms_exp;
+ms_exp.setSpectra(ms_spectra);
+
+//construct MSExperiment without m2 level
+MSExperiment ms1_exp;
+ms1_exp.setSpectra(ms1_spectra);
+
+//construct MSExperiment with two m2 level
+MSExperiment ms2_2_exp;
+ms2_2_exp.setSpectra(ms2_2_spectra);
+
+//construct empty MSExperiment
+MSExperiment ms_empty_exp;
+
+
+
+//////////////////////////////////////////////////////////////////
+//start Section
+/////////////////////////////////////////////////////////////////
+
+Ms2IdentificationRate* ptr = nullptr;
+Ms2IdentificationRate* nulpt = nullptr;
+START_SECTION(Ms2IdentificationRate())
+{
+  ptr = new Ms2IdentificationRate();
+  TEST_NOT_EQUAL(ptr, nulpt)
+}
+END_SECTION
+
+START_SECTION(~Ms2IdentificationRate())
+{
+  delete ptr;
+}
+END_SECTION
+
+
+Ms2IdentificationRate ms2ir;
+Ms2IdentificationRate ms2ir_fdr;
+Ms2IdentificationRate ms2ir_ms1;
+Ms2IdentificationRate ms2ir_ms2_2;
+Ms2IdentificationRate ms2ir_empty_msexp;
+Ms2IdentificationRate ms2ir_empty_fmap;
+
+//tests compute function
+START_SECTION(void Ms2IdentificationRate::compute(FeatureMap const & feature_map, MSExperiment const & exp, bool force_fdr = false))
+{
+  //test with valid input
+  ms2ir.compute(fmap, ms_exp);
+  std::vector<Ms2IdentificationRate::IdentificationRateData> result;
+  result = ms2ir.getResults();
+
+  for (auto idrd : result)
+  {
+    TEST_EQUAL(idrd.num_peptide_identification, 2)
+    TEST_EQUAL(idrd.num_ms2_spectra, 6)
+    TEST_REAL_SIMILAR(idrd.identification_rate, (double)1/3)
+  }
+
+  //less ms2 spectra than identifictions
+  TEST_EXCEPTION_WITH_MESSAGE(Exception::Precondition, ms2ir_ms2_2.compute(fmap, ms2_2_exp), "There are more Identifications than Ms2-Level. Please check your data.")
+
+  //empty ms experiment
+  TEST_EXCEPTION_WITH_MESSAGE(Exception::MissingInformation, ms2ir_empty_msexp.compute(fmap, ms_empty_exp), "mzML is corrupted or empty")
+
+  //empty feature map
+  TEST_EXCEPTION_WITH_MESSAGE(Exception::MissingInformation, ms2ir_empty_fmap.compute(fmap_empty, ms_exp), "FeatureXML is corrupted or empty")
+
+  //no fdr
+  TEST_EXCEPTION_WITH_MESSAGE(Exception::Precondition, ms2ir_fdr.compute(fmap_fdr, ms_exp), "FDR was not made. If you want to continue without FDR use -force_fdr")
+
+  //no ms2 spectra
+  TEST_EXCEPTION_WITH_MESSAGE(Exception::MissingInformation, ms2ir_ms1.compute(fmap, ms1_exp), "No MS2-Level found")
+
```

Comment:
add a test for force_fdr == true, i.e. meta value should be ignored (you can use the same data).